pkg/volmount/volmount_test.go

```diff
@@ -0,0 +1,245 @@
+	if err != nil {
+		if errno, ok := errors.Cause(err).(syscall.Errno); ok && errno == syscall.ERROR_PRIVILEGE_NOT_HELD {
+			t.Skip(err)
+		}
+		t.Fatal(err)
+	}
+	defer func() {
+		detachErr := vhd.DetachVirtualDisk(vhdHandle)
+		if detachErr != nil {
+			// assuming if we're already failing, failing more isn't wrong
+			t.Fatal(detachErr)
+		}
+	}()
+
+	physPath, err := vhd.GetVirtualDiskPhysicalPath(vhdHandle)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	volumePath, err := turnRawDiskIntoAVolume(physPath)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	mountPoints, err := GetVolumePathNamesForVolumeName(volumePath)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if len(mountPoints) != 0 {
+		t.Fatalf("Brand new volume was unexpectedly mounted at: %v", mountPoints)
+	}
+
+	mountPoint0 := filepath.Join(dir, "mount0")
+	err = os.MkdirAll(mountPoint0, 0)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = SetVolumeMountPoint(mountPoint0, volumePath)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	defer func() {
+		deleteErr := DeleteVolumeMountPoint(mountPoint0)
+		if deleteErr != nil {
+			// Assuming if we're already failing, failing more isn't wrong.
+			t.Fatal(deleteErr)
+		}
+	}()
+
+	mountPoint0VolumePath, err := GetVolumeNameForVolumeMountPoint(mountPoint0)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if mountPoint0VolumePath != volumePath {
+		t.Fatalf("Mount 0 read-back incorrectly, expected %s; got %s", volumePath, mountPoint0VolumePath)
+	}
+
+	mountPoints, err = GetVolumePathNamesForVolumeName(volumePath)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	expectedMountPoints := []string{mountPoint0 + string(filepath.Separator)}
+	require.ElementsMatch(t, expectedMountPoints, mountPoints, "Mount apparently failed, expected %v (order irrelevant): got %v", expectedMountPoints, mountPoints)
+
+	mountPoint1 := filepath.Join(dir, "mount1")
+	err = os.MkdirAll(mountPoint1, 0)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = SetVolumeMountPoint(mountPoint1, volumePath)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	defer func() {
+		deleteErr := DeleteVolumeMountPoint(mountPoint1)
+		if deleteErr != nil {
+			// Assuming if we're already failing, failing more isn't wrong.
+			t.Fatal(deleteErr)
+		}
+	}()
+
+	mountPoint1VolumePath, err := GetVolumeNameForVolumeMountPoint(mountPoint1)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if mountPoint1VolumePath != volumePath {
+		t.Fatalf("Mount 1 read-back incorrectly, expected %s; got %s", volumePath, mountPoint1VolumePath)
+	}
+
+	mountPoints, err = GetVolumePathNamesForVolumeName(volumePath)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// No order guarantee on the mounts
+	expectedMountPoints = []string{mountPoint0 + string(filepath.Separator), mountPoint1 + string(filepath.Separator)}
+	require.ElementsMatch(t, expectedMountPoints, mountPoints, "Mount apparently failed, expected %v (order irrelevant): got %v", expectedMountPoints, mountPoints)
+
+	// Note: DeleteVolumeMountPoint is tested in the defers above.
+}
```

Comment:
That's correct, it is the same thing twice, because I'm validating the 'return list of mount points'. I can easily move that to a helper function, I think my rationale at the time was "I should call the things I'm testing directly in the test where possible", but I'm not _beholden_ to that.